### PR TITLE
*: add parser support for multiple table renaming

### PR DESCRIPTION
**Pull request by holys**
Why make such changes?

One of our important clients want to support case like `rename table t1 to t3, t2 to t4` in syncer, and syncer is heavily depends on parser.

1. It won't affect single table renaming;
2. For multiple table renaming, parser works, but it will return error like `alter table` does.
> ERROR 1105 (HY000): can't run multi schema change

 PTAL @shenli  @zimulala  @tiancaiamao

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: ast/ddl.go [hunk 1]
```diff
@@ -469,6 +469,9 @@ type RenameTableStmt struct {
 
 	OldTable *TableName
 	NewTable *TableName
+	// TODO: refactor this when you are going to add full support for multiple schema changes.
```
Comment (zimulala) [inline ast/ddl.go]: s/refactor/Refactor

File: ast/ddl.go [hunk 2]
```diff
@@ -488,6 +491,42 @@ func (n *RenameTableStmt) Accept(v Visitor) (Node, bool) {
...
+		if !ok {
+			return n, false
+		}
+		n.TableToTables[i] = node.(*TableToTable)
+	}
+
+	return v.Leave(n)
+}
+
+// TableToTable represents renaming old table to new table used in RenameTableStmt
```
Comment (zimulala) [inline ast/ddl.go]: Add `.` at the end of a line.

File: parser/parser.y [hunk 3]
```diff
@@ -1158,6 +1164,34 @@ RenameTableStmt:
 			NewTable: $5.(*ast.TableName),
 		}
 	 }
+|	"RENAME" "TABLE" TableToTableList ',' TableToTable
```
Comment (zimulala) [inline parser/parser.y]: Is it enough to use `TableToTableList`?

File: executor/ddl.go [hunk 4]
```diff
@@ -103,6 +103,10 @@ func (e *DDLExec) executeTruncateTable(s *ast.TruncateTableStmt) error {
 }
 
 func (e *DDLExec) executeRenameTable(s *ast.RenameTableStmt) error {
+	if len(s.TableToTables) != 0 {
```
Comment (zimulala) [inline executor/ddl.go]: Add a test case in `ddl_db_test.go`.

Comment (holys) [inline executor/ddl.go]: @zimulala  PTAL again. Test case added.

File: executor/ddl.go [hunk 5]
```diff
@@ -103,6 +103,10 @@ func (e *DDLExec) executeTruncateTable(s *ast.TruncateTableStmt) error {
 }
 
 func (e *DDLExec) executeRenameTable(s *ast.RenameTableStmt) error {
+	if len(s.TableToTables) != 1 {
+		// Now we only allow one schema changing at the same time.
+		return errors.Errorf("can't run multi schema change")
```
Comment (zimulala) [inline executor/ddl.go]: Use terror instead of this error.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**